scapy/utils.py

```diff
@@ -664,7 +676,104 @@ def read_all(self,count=-1):
         return plist.PacketList(res,name = os.path.basename(self.filename))
     def recv(self, size=MTU):
         return self.read_packet(size=size)
-        
+
+
+class RawPcapNgReader(RawPcapReader):
+    """A stateful pcapng reader. Each packet is returned as a
+    string.
+
+    """
+
+    def __init__(self, filename):
+        self.filename = filename
+        # A list of (linktype, snaplen); will be populated by IDBs.
+        self.interfaces = []
+        self.blocktypes = {
+            1: self.read_block_idb,
+            6: self.read_block_epb,
+        }
+        try:
+            self.f = gzip.open(filename,"rb")
+            magic = self.f.read(4)
+        except IOError:
+            self.f = open(filename,"rb")
+            magic = self.f.read(4)
+        if magic != "\n\r\r\n": # PcapNg:
```

Comment:
I guess that \x0a\x0d\x0d\x0a will be better to stick with the specs. Also renaming it block_type might ease future modifications.